app/src/main/java/ch/epfl/sdp/peakar/utils/OnSwipeTouchListener.java

```diff
@@ -0,0 +1,148 @@
+package ch.epfl.sdp.peakar.utils;
+
+import android.content.Context;
+import android.content.Intent;
+import android.util.DisplayMetrics;
+import android.view.GestureDetector;
+import android.view.MotionEvent;
+import android.view.View;
+
+import ch.epfl.sdp.peakar.camera.CameraActivity;
+import ch.epfl.sdp.peakar.gallery.GalleryActivity;
+import ch.epfl.sdp.peakar.general.SettingsActivity;
+import ch.epfl.sdp.peakar.map.MapActivity;
+import ch.epfl.sdp.peakar.social.SocialActivity;
+
+/**
+ * Detects left and right swipes across a view.
+ */
+public class OnSwipeTouchListener implements View.OnTouchListener {
```

Comment:
I also see that if you swipe when on the borders (settings, social) that you get to the camera activity. shouldn't it be that you can't swipe at the border?

Reply:
I made it that when you are on the outer activities (Settings and Social) and swipe out, that you return to the camera-preview (like a shortcut) but this can be change as we wish to